docs/src/guide/invariants/index.md

```diff
@@ -0,0 +1,330 @@
+# {#guide-invariants} Writing Invariants
+
+Invariants are an essential feature when writing complex Reach DApps. 
+Our guide @{seclink("guide-loop-invs")} provides a deep theoretical understanding of loop invariants, and the @{seclink("tut")} and @{seclink("tut-rsvp")} tutorials both feature invariants. If you desire to create a complex Reach DApp, then you'll likely need to write your own invariants, as well.
+
+To help prepare you to write your own invariants, we'll review a few examples and break them to help understand the error messages. 
+At the end of the guide you'll have a respect for how important invariants are in Reach Applications and how important it is to have a complete understanding of your application before writing the codebase. 
+
+First, a few basics about invariants. 
+
+Invariant refers to something that is constant without variation. 
+
+Loop invariants are required when writing Reach `{!rsh} while` loops.
+They are established immediately before the loop begins. 
+You are allowed to write as many `{!rsh} invariant` statements as needed. 
+Inside a `{!rsh} parallelReduce`, use dot notation to place each `{!rsh} invariant` on its own line to easily distinguish each `{!rsh} invariant` from the other.
+
+The `{!rsh} while` loop condition and the body of the loop must uphold the `{!rsh} invariant`. 
+To expand, there are moments within the loop that the invariant may be invalidated, but by the end of the loop, the invariant must be true. 
+
+The invariant establishes properties for the rest of the program after the loop ends.
+So the invariants establish truth just before the loop, uphold that truth inside the loop, and remains true after the loop ends.
+
+Let's look at some examples of invariants and break them to understand their importance. 
+
+``` rsh
+load: /examples/rps-7-loops/index.rsh
+md5: ee287e712cdfe8d91bbb038c383d25d3
+range: 59-61
+```
+
+This is from the @{seclink("tut")} tutorial. 
+The while loop features two invariants. 
+The first controls the balance in relation to the wager amount and the second ensures that the outcome is one of the enumerated outcomes from the top of the program.
+
+In this example, if the balance ever becomes something other than double the amount of the wager then the verification engine will throw an error. 
```

Comment:
This makes it sound like this could happen. It is more than if the programmer ever modified the program in a way where it wasn't guaranteed, there would be an error.